### PR TITLE
Add items to standard logs

### PR DESCRIPTION
**Pull request by Joshi234**
This is useful if for troubleshooting issues.

Closes https://github.com/Ryujinx/Ryujinx/issues/1728

### CODE REVIEW

File: Ryujinx/Ui/MainWindow.cs [hunk 1]
```diff
@@ -451,6 +451,9 @@ public void LoadApplication(string path)
                     }
                 }
 
+                Logger.Notice.Print(LogClass.Application, $"AudioBackend: {ConfigurationState.Instance.System.AudioBackend.Value}");
```
Comment (AcK77) [inline Ryujinx/Ui/MainWindow.cs:454]: Those logs should be done when the users change settings too (with Toggles or in the Settings window).

File: Ryujinx/Ui/Windows/SettingsWindow.cs [hunk 2]
```diff
@@ -424,7 +433,10 @@ private void SaveSettings()
 
             if (_audioBackendSelect.GetActiveIter(out TreeIter activeIter))
             {
-                ConfigurationState.Instance.System.AudioBackend.Value = (AudioBackend)_audioBackendStore.GetValue(activeIter, 1);
+                if((AudioBackend)_audioBackendStore.GetValue(activeIter, 1)!= ConfigurationState.Instance.System.AudioBackend.Value) {
```
Comment (Thog) [inline Ryujinx/Ui/Windows/SettingsWindow.cs:436]:
```suggestion
                if((AudioBackend)_audioBackendStore.GetValue(activeIter, 1) != ConfigurationState.Instance.System.AudioBackend.Value)
                {
```

Comment (Thog) [reply Ryujinx/Ui/Windows/SettingsWindow.cs:436]: Also could be interesting to use a temporary variable for `` (AudioBackend)_audioBackendStore.GetValue(activeIter, 1);``

**Review by Thog — CHANGES_REQUESTED** [review]
Just small nits, otherwise lgtm

**Comment by mageven** [comment]
Related issue: #1728. Toggle events are missing I suppose. Could be worth adding.

**Comment by Joshi234** [comment]
> Related issue: #1728. Toggle events are missing I suppose. Could be worth adding.

Toggle Events doesn't make any sense when they dont get saved, that's why it's only logged when saving

**Comment by mageven** [comment]
What happens if a crash occurs when they toggled Vsync or Docked mode?

**Comment by Joshi234** [comment]
> What happens if a crash occurs when they toggled Vsync or Docked mode?

Missed that you can click on the text, gonna add that

File: Ryujinx/Ui/MainWindow.cs [hunk 3]
```diff
@@ -827,11 +830,13 @@ private void Row_Activated(object sender, RowActivatedArgs args)
         private void VSyncStatus_Clicked(object sender, ButtonReleaseEventArgs args)
         {
             _emulationContext.EnableDeviceVsync = !_emulationContext.EnableDeviceVsync;
+            Logger.Notice.Print(LogClass.Application, $"VSync changed to: {_emulationContext.EnableDeviceVsync}");
```
Comment (AcK77) [inline Ryujinx/Ui/MainWindow.cs:833]:
```suggestion
            Logger.Notice.Print(LogClass.Application, $"VSync toggled to: {_emulationContext.EnableDeviceVsync}");
```
and add a new empty line before

File: Ryujinx/Ui/MainWindow.cs [hunk 4]
```diff
@@ -827,11 +830,13 @@ private void Row_Activated(object sender, RowActivatedArgs args)
...
         }
 
         private void DockedMode_Clicked(object sender, ButtonReleaseEventArgs args)
         {
             ConfigurationState.Instance.System.EnableDockedMode.Value = !ConfigurationState.Instance.System.EnableDockedMode.Value;
+            Logger.Notice.Print(LogClass.Application, $"IsDocked changed to: {ConfigurationState.Instance.System.EnableDockedMode.Value}");
```
Comment (AcK77) [inline Ryujinx/Ui/MainWindow.cs:839]:
```suggestion
            Logger.Notice.Print(LogClass.Application, $"IsDocked toggled to: {ConfigurationState.Instance.System.EnableDockedMode.Value}");
```
and add a new empty line before

File: Ryujinx/Ui/Windows/SettingsWindow.cs [hunk 5]
```diff
@@ -389,6 +390,14 @@ private void SaveSettings()
             {
                 ConfigurationState.Instance.System.TimeZone.Value = _systemTimeZoneEntry.Text;
             }
+            if(ConfigurationState.Instance.Graphics.EnableVsync.Value != _vSyncToggle.Active)
```
Comment (AcK77) [inline Ryujinx/Ui/Windows/SettingsWindow.cs:393]: Add a new empty line before

File: Ryujinx/Ui/Windows/SettingsWindow.cs [hunk 6]
```diff
@@ -389,6 +390,14 @@ private void SaveSettings()
...
+            {
+                Logger.Notice.Print(LogClass.Application, $"Vsync changed to: {_vSyncToggle.Active}");
+            }
+            if(ConfigurationState.Instance.System.EnableDockedMode.Value != _dockedModeToggle.Active)
```
Comment (AcK77) [inline Ryujinx/Ui/Windows/SettingsWindow.cs:397]: Add a new empty line before

File: Ryujinx/Ui/Windows/SettingsWindow.cs [hunk 7]
```diff
@@ -424,7 +433,11 @@ private void SaveSettings()
 
             if (_audioBackendSelect.GetActiveIter(out TreeIter activeIter))
             {
-                ConfigurationState.Instance.System.AudioBackend.Value = (AudioBackend)_audioBackendStore.GetValue(activeIter, 1);
+                AudioBackend audioBackend = (AudioBackend)_audioBackendStore.GetValue(activeIter, 1);
+                if (audioBackend != ConfigurationState.Instance.System.AudioBackend.Value) {
```
Comment (AcK77) [inline Ryujinx/Ui/Windows/SettingsWindow.cs:437]:
```suggestion
                if (audioBackend != ConfigurationState.Instance.System.AudioBackend.Value)
                { 
```

File: Ryujinx/Ui/Windows/SettingsWindow.cs [hunk 8]
```diff
@@ -424,7 +433,11 @@ private void SaveSettings()
 
             if (_audioBackendSelect.GetActiveIter(out TreeIter activeIter))
             {
-                ConfigurationState.Instance.System.AudioBackend.Value = (AudioBackend)_audioBackendStore.GetValue(activeIter, 1);
+                AudioBackend audioBackend = (AudioBackend)_audioBackendStore.GetValue(activeIter, 1);
+                if (audioBackend != ConfigurationState.Instance.System.AudioBackend.Value) { 
+                    ConfigurationState.Instance.System.AudioBackend.Value = audioBackend;
+                    Logger.Notice.Print(LogClass.Application, $"AudioBackend changed to: {audioBackend}");
```
Comment (AcK77) [inline Ryujinx/Ui/Windows/SettingsWindow.cs:439]: Add a new empty line before

**Review by AcK77 — CHANGES_REQUESTED** [review]

**Review by riperiperi — APPROVED** [review]
Nice, this is helpful.

File: Ryujinx/Ui/MainWindow.cs [hunk 9]
```diff
@@ -827,11 +830,13 @@ private void Row_Activated(object sender, RowActivatedArgs args)
         private void VSyncStatus_Clicked(object sender, ButtonReleaseEventArgs args)
         {
             _emulationContext.EnableDeviceVsync = !_emulationContext.EnableDeviceVsync;
+            Logger.Notice.Print(LogClass.Application, $"VSync toggled to: {_emulationContext.EnableDeviceVsync}");
```
Comment (AcK77) [inline Ryujinx/Ui/MainWindow.cs:833]: Add a new empty line before.

File: Ryujinx/Ui/MainWindow.cs [hunk 10]
```diff
@@ -827,11 +830,13 @@ private void Row_Activated(object sender, RowActivatedArgs args)
...
         }
 
         private void DockedMode_Clicked(object sender, ButtonReleaseEventArgs args)
         {
             ConfigurationState.Instance.System.EnableDockedMode.Value = !ConfigurationState.Instance.System.EnableDockedMode.Value;
+            Logger.Notice.Print(LogClass.Application, $"IsDocked toggled to: {ConfigurationState.Instance.System.EnableDockedMode.Value}");
```
Comment (AcK77) [inline Ryujinx/Ui/MainWindow.cs:839]: Add a new empty line before.

File: Ryujinx/Ui/Windows/SettingsWindow.cs [hunk 11]
```diff
@@ -390,6 +391,16 @@ private void SaveSettings()
                 ConfigurationState.Instance.System.TimeZone.Value = _systemTimeZoneEntry.Text;
             }
 
+            if(ConfigurationState.Instance.Graphics.EnableVsync.Value != _vSyncToggle.Active)
```
Comment (AcK77) [inline Ryujinx/Ui/Windows/SettingsWindow.cs:394]:
```suggestion
            if (ConfigurationState.Instance.Graphics.EnableVsync.Value != _vSyncToggle.Active)
```

File: Ryujinx/Ui/Windows/SettingsWindow.cs [hunk 12]
```diff
@@ -390,6 +391,16 @@ private void SaveSettings()
...
+            {
+                Logger.Notice.Print(LogClass.Application, $"Vsync changed to: {_vSyncToggle.Active}");
+            }
+
+            if(ConfigurationState.Instance.System.EnableDockedMode.Value != _dockedModeToggle.Active)
```
Comment (AcK77) [inline Ryujinx/Ui/Windows/SettingsWindow.cs:399]:
```suggestion
            if (ConfigurationState.Instance.System.EnableDockedMode.Value != _dockedModeToggle.Active)
```

**Review by AcK77 — CHANGES_REQUESTED** [review]

**Review by mageven — COMMENTED** [review]
The current logs for toggles are a bit inconsistent. Vsync can also be toggled via hotkeys but it's not logged; and changes to Vsync inside Settings Window don't affect the current run. 
Also, I prefer logs at places where config is actually set rather than `LoadApplication()` and Settings Window.

So my suggestions- since they only affect emulation, a pair of "set" and "changed" logs.

"set" for both- `Initialize()` in `Switch.cs` is a good location for now.
Docked "changed" - `OnDockedModeChange()` in `Horizon.cs`
Vsync "changed" - It'd be nice if a common method was introduced like the one for docked. Otherwise, one at every toggle location.

**Comment by Joshi234** [comment]
Messed up everything gonna do a new pr

**Comment by Joshi234** [comment]
I changed the locations of the stuff as @mageven suggested.

File: Ryujinx/Ui/Windows/SettingsWindow.cs [hunk 13]
```diff
@@ -396,6 +397,11 @@ private void SaveSettings()
                 ConfigurationState.Instance.System.TimeZone.Value = _systemTimeZoneEntry.Text;
             }
 
+            if (ConfigurationState.Instance.Graphics.EnableVsync.Value != _vSyncToggle.Active)
+            {
+                Logger.Notice.Print(LogClass.Application, $"Vsync toggled to: {_vSyncToggle.Active}");
```
Comment (mageven) [inline Ryujinx/Ui/Windows/SettingsWindow.cs:402]: I think this isn't needed. This option comes into effect the next time emulation is started and there's already a log for that.

**Review by mageven — COMMENTED** [review]
Looks much better now. I forgot to mention this previously but `Logger.Info` is better suited here than `Logger.Notice` for all of these.

File: Ryujinx/Ui/Windows/SettingsWindow.cs (hunk 13 above)
Comment (Joshi234) [reply Ryujinx/Ui/Windows/SettingsWindow.cs:402]: You could change the setting while a game is running tho

Comment (Joshi234) [reply Ryujinx/Ui/Windows/SettingsWindow.cs:402]: nevermind it doesnt effect the running game

**Comment by Joshi234** [comment]
This is ready for review btw

**Review by AcK77 — APPROVED** [review]

**Review by Thog — APPROVED** [review]
lgtm

**Pull request merged**